### PR TITLE
[DSL] Introduce new endpoint to expose data stream lifecycle stats

### PR DESCRIPTION
**Pull request by gmarouli**
We introduce the endpoint `GET _lifecycle/stats` that exposes stats from the data stream lifecycle run and the data streams and backing indices managed by DSL. For example:

```
GET _lifecycle/stats?human&pretty
{
  "last_run_duration_in_millis": 2,
  "last_run_duration": "2ms",
  "time_between_starts_in_millis": 9998,
  "time_between_starts": "9.99s",
  "data_streams_count": 2,
  "data_streams": [
    {
      "name": "my-other-stream",
      "backing_indices_in_total": 2,
      "backing_indices_in_error": 1
    },
    {
      "name": "my-data-stream",
      "backing_indices_in_total": 2,
      "backing_indices_in_error": 0
    }
  ]
}
```

### CODE REVIEW

**Comment by gmarouli** [comment]
@elasticmachine update branch

File: modules/data-streams/src/internalClusterTest/java/org/elasticsearch/datastreams/lifecycle/DataStreamLifecycleServiceIT.java [hunk 1]
```diff
@@ -706,6 +677,35 @@ public void testReenableDataStreamLifecycle() throws Exception {
         });
     }
 
+    private static List<String> getBackingIndices(String dataStreamName) {
```
Comment (gmarouli) [inline modules/data-streams/src/internalClusterTest/java/org/elasticsearch/datastreams/lifecycle/DataStreamLifecycleServiceIT.java:680]: I just moved this helper function in the bottom to be after all the test cases.

**Comment by gmarouli** [comment]
@elasticmachine update branch

File: docs/reference/data-streams/lifecycle/apis/get-lifecycle-stats.asciidoc [hunk 2]
```diff
@@ -0,0 +1,93 @@
+[[data-streams-get-lifecycle-stats]]
+=== Get data stream lifecycle stats
```
Comment (andreidan) [inline docs/reference/data-streams/lifecycle/apis/get-lifecycle-stats.asciidoc:2]: Very nice ! ❤️

File: modules/data-streams/src/internalClusterTest/java/org/elasticsearch/datastreams/lifecycle/DataStreamLifecycleServiceIT.java (hunk 1 above)
Comment (andreidan) [reply modules/data-streams/src/internalClusterTest/java/org/elasticsearch/datastreams/lifecycle/DataStreamLifecycleServiceIT.java:674]: Thanks for this note, this is really helpful when reviewing ❤️

File: modules/data-streams/src/main/java/org/elasticsearch/datastreams/lifecycle/action/GetDataStreamLifecycleStatsAction.java [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+
+        public Request() {}
+
+        @Override
+        public ActionRequestValidationException validate() {
+            return null;
+        }
+    }
+
+    public static class Response extends ActionResponse implements ChunkedToXContentObject {
```
Comment (andreidan) [inline modules/data-streams/src/main/java/org/elasticsearch/datastreams/lifecycle/action/GetDataStreamLifecycleStatsAction.java:54]: Chunked response ❤️

**Review by andreidan — APPROVED** [review]
Thanks for adding this Mary, this is really nice ! 🚀 

LGTM

**Comment by gmarouli** [comment]
@elasticmachine update branch

**Pull request merged**